### PR TITLE
fix(agent): apply local-skill context on the codex adapter

### PR DESCRIPTION
**Pull request by tatoalo**
## Problem

Local skills didn't actually apply on the Codex adapter. The agent-server sets `_meta.localSkillContext` (the resolved skill instructions) and leaves the raw `/skill` slash command in the prompt; the Claude adapter consumes that meta, dropping the `/skill` chunk and injecting the instructions — but the Codex adapter had no such seam, so the bare `/skill` reached codex-acp and the skill instructions were ignored.

## Changes

- added `prependLocalSkillContext` to the Codex adapter, mirroring the Claude adapter's `promptToClaude` handling: drop the leading `/skill` chunk (the context already carries the user's args), prepend the resolved skill instructions as a text block, and strip the local-skill `_meta` before forwarding to codex-acp
- extracted the shared `isLocalSkillCommandChunk` helper into `adapters/local-skill.ts`, now imported by both the Codex and Claude adapters (previously duplicated)

### CODE REVIEW

**Comment by tatoalo** [comment]
* **#3058** <a href="https://app.graphite.com/github/pr/PostHog/code/3058?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/PostHog/code/3058?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#3057** <a href="https://app.graphite.com/github/pr/PostHog/code/3057?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3056** <a href="https://app.graphite.com/github/pr/PostHog/code/3056?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/agent/src/adapters/codex/codex-agent.ts [hunk 1]
```diff
@@ -168,6 +169,49 @@ function prependPrContext(params: PromptRequest): PromptRequest {
...
+    if (
+      !skipped &&
+      localSkillName &&
+      isLocalSkillCommandChunk(chunk, localSkillName)
+    ) {
+      skipped = true;
+      return false;
+    }
+    return true;
+  });
```
[the remark this answers is not in the document]
Comment (tatoalo) [reply packages/agent/src/adapters/codex/codex-agent.ts:204]: Fixed — added an explicit guard: if `localSkillContext` is set but `localSkillName` is absent, the function now returns the prompt unchanged rather than injecting context while leaving the bare `/skill` chunk in place, so we never emit the broken context-plus-stray-command mix.

File: packages/agent/src/adapters/codex/codex-agent.test.ts [hunk 2]
```diff
@@ -313,6 +313,48 @@ describe("CodexAcpAgent", () => {
...
+    expect(forwarded._meta?.localSkillName).toBeUndefined();
+    // The broadcast still shows the real user turn (the typed command).
+    expect(client.sessionUpdate).toHaveBeenCalledWith({
+      sessionId: "session-1",
+      update: {
+        sessionUpdate: "user_message_chunk",
+        content: { type: "text", text: "/depparent run the readiness check" },
+      },
+    });
+  });
```
[the remark this answers is not in the document]
Comment (tatoalo) [reply packages/agent/src/adapters/codex/codex-agent.test.ts:356]: Done — parameterized the test with the two edge cases (missing `localSkillName`, and no `localSkillContext`), each asserting the prompt is forwarded to codex-acp unchanged.
